app/models/task.rb

```diff
@@ -9,13 +9,17 @@ class Task < ActiveRecord::Base
   # has_many :requirements
   # has_many :assignments, through: :requirements
 
-  STATUS_CHOICES_ARRAY = ['Empty', 'Partially Filled', 'Full', 'Cancelled']
-
   # The following provides the equivalent of:
-  # STATUS_CHOICES = { 'Empty' => 0, 'Partially Filled' => 1, 'Full' => 2, 'Cancelled' => 3 }
-  # This is used to give a sortable numeric value based on the order of the above array.
+  # STATUS_CHOICES = { 'Empty' => 0, 'Inadequate' => 1, 'Adequate' => 2, 'Satisfied' => 3, 'Full' => 4, 'Cancelled' => 5 }
+  STATUS_CHOICES_ARRAY = ['Empty', 'Inadequate', 'Adequate', 'Satisfied', 'Full', 'Cancelled']
   STATUS_CHOICES = STATUS_CHOICES_ARRAY.map.with_index { |v, i| [v, i] }.to_h
```

Comment:
Still not sure about this line. I'll merge it, but it seems unnecessarily complex. You could
```
STATUS_CHOICES = { 'Empty' => 0, 'Inadequate' => 1, 'Adequate' => 2, 'Satisfied' => 3, 'Full' => 4, 'Cancelled' => 5 }
STATUS_CHOICES_ARRAY=STATUS_CHOICES.keys
```
Wouldn't that work ?